File: solution/78/answer.cpp
#include <stdio.h>
#include <vector>

class Solution {
public:
    std::vector<std::vector<int> > subsets(std::vector<int>& nums) {
        int allnum = 1 << nums.size();
        std::vector<std::vector<int> > result;
        for(int i=0;i< allnum;i++){
            std::vector<int> item;
            for(int x = 0;x< nums.size();x++){
                int binx = 1 << x;
                if(binx&i){
                    item.push_back(nums[x]);
                }
            }
            result.push_back(item);
        }
        return result;
    }
};

File: solution/78/answer.py
class Solution(object):

    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        allnum = 1 << len(nums)
        result = []
        for i in xrange(allnum):
            item = []
            for x in xrange(len(nums)):
                if (1 << x) & i:
                    item.append(nums[x])
            result.append(item)
        return result


s = Solution()
print s.subsets([1,2,3])